### DISCUSSION TITLE
Cracking noise

### DISCUSSION THREAD

**Discussion by oliverk71** [mcguirepr89/BirdNET-Pi#173] [Q&A]
I have some cracking noise on my recordings, but I do not know where it comes from. Anyone who has the same problem or know what can be the cause? 

![image](https://user-images.githubusercontent.com/10213957/160209653-3dbd6d68-b84e-460f-b95b-735afd5125a6.png)

Here is another one:
![image](https://user-images.githubusercontent.com/10213957/160210179-3ab09f87-9468-47ed-9b7a-ad6d1a236e31.png)
and the referring recording:
https://birdnetgv.ddnss.de/By_Date/2022-03-25/Kohlmeise/Kohlmeise-81%25-2022-03-25-birdnet-09:36:59.mp3

It does not matter so much, but if possible, I would like to get rid of it. Any ideas?

I use this electret microphone capsule: AOM-5024L-HD-R
And this USB soundcard:  
And a non-original power supply. If no one has an idea, I would maybe change the power supply.

**Comment by mcguirepr89** [comment]
Sorry, Oliver -- no ideas from me :/

**Reply by pjalocha** [reply]
I see exactly same, when I connected a "good" microphone with 192kHz sampling rate, the corresponding messages about issues with USB transfer can be seen in dmesg - I need yet to fight it.

I can confirm, the problem is gone, when I switched to Rapsberry PI 4, otherwise I was not able to solve it.

The problem actually started when I purchased a microphone with 192kHz sampling rate and 24-bit thus produce much higher data rate. But I must say I still don't really understand it, because like RTLSDR receivers produce higher data rates and have no issues.

**Comment by DD4WH** [comment]
Oliver, I had this very same problem when I used a Raspberry Pi 3B+ for BirdNET-Pi. 
I then tested the very same microphone setup and it worked without any crackling noise on a Raspberry Pi 4.
I am not sure whether this means, that my 3B+ had an issue or the crackling noise is present on EVERY Raspyberry Pi 3B+. I did not investigate this systematically.

P.S.: is that Redwing in the first recording? Nice migrant catch . . .

**Reply by oliverk71** [reply]
Interesting! Because I use a 3B. But I think, I will exchange the power supply first, then try a 4B. But I can't try it this week, unfortunately I am very busy.
Yeah, it seems to be a redwing. I have a few recordings of a redwing in march. If you want to have a look: https://birdnetgv.ddnss.de/By_Common_Name/Rotdrossel/

**Comment by nilspupils** [comment]
Some more people have noticed cracking on the RPi3. I found this info about turning off timer-based audio scheduling. 
https://fedoraproject.org/wiki/How_to_debug_PulseAudio_problems

Also the "suspend-on-idle" function might be relevant: 
https://wiki.archlinux.org/title/PulseAudio/Troubleshooting#Pops_when_starting_and_stopping_playback

Last, one comment here about disabling LAN and connecting through WLAN (because LAN and USB share the same bus).
https://community.roonlabs.com/t/raspberry-pi-3b-clicking-on-usb-audio-devices/45597/3

Please report back if that helps.

**Reply by VictorioBerra** [reply]
I disabled that pulse audio module, and I set my eth card to down. Same problem.

Gonna try and find a Pi 4 when stock returns.

**Reply by DD4WH** [reply]
I experienced similar noise with the RPi3B. Could you try this:
* have a Look into System Info
* scroll down to "throttling"
* if throttling occurred due to low voltage, try a more powerful supply
* I powered the Pi3 over the GPIO pins, because it seemed that the micro-USB cables I used were not capable of delivering enough current, although the wall plugs'  specs did say so
* after power supply was sufficient==no more throttling, the crackling noise was significantly reduced
* with the same mic and the Pi4, I experienced no crackling at all

Hope that helps, Regards, Frank DD4WH

**Reply by oliverk71** [reply]
I suspected the power supply, but unfortunately did not find any time to check it, by now, Hope, I can test this and other this or next week. 
Frank, where can I find "throttling" in System Info? Maybe I cannot see it because my system did not throttle?
However, I have this crackling noise and will try the different options suggested by the users.

**Comment by oliverk71** [comment]
Meanwhile I changed the power supply, but without any effect. 

I wonder if the USB soundcard is used for output, too. Or if internal sound is used. Because I found this discussion: https://forum.openframeworks.cc/t/audio-crackling-in-raspberry-pi-3b/33836

**Reply by oliverk71** [reply]
Changed the audio card in the settings and since then: no more crackling noises.

**Reply by lazydave1** [reply]
Hi Oliver,

i had the same problem as you, and for me changing the audio card also worked for getting rid of the crackling noise, but now it won't analyze the files anymore... 
I can listen live without any problems, but it doesn't show anything new.
To what did you change your settings?

Kind regards
David

**Reply by ampsu** [reply]
Yeah, it is weird. When I change the audio card, the crackling stops but also analyzing stops. So looks like it is a Pulse problem but I have no idea why recording also stops.

**Comment by pjsg** [comment]
I moved from a 3B to a 4 to fix this. I was getting alarming messages in the `dmesg` output on the 3B that seemed to be USB problems. These don't happen on the 4.
